packages/lib/models/Setting.ts

```diff
@@ -1975,6 +1975,8 @@ class Setting extends BaseModel {
 		if (md.type === SettingItemType.Array) {
 			if (!value) return [];
 			if (Array.isArray(value)) return value;
+			// Hack to transform existing string values of spellChecker.language to array.
+			if (key === 'spellChecker.language' && typeof value === 'string') return [value];
```

Comment:
Can't really use `applyDefaultMigrations`, since it's desined to change user setting, only if the user never touched the setting before.

Reply:
> only if the user never touched the setting before.

Adding a new setting could address this (e.g. `languages` instead of `language`).

(Having two settings (one of which is unused) might be confusing, however. As such, I'm not sure whether or not I like the solution I suggested...)